src/main/java/org/broadinstitute/consent/http/service/LibraryCardService.java

```diff
@@ -27,10 +29,15 @@ public LibraryCardService(LibraryCardDAO libraryCardDAO, InstitutionDAO institut
         this.userDAO = userDAO;
     }
 
-    public LibraryCard createLibraryCard(LibraryCard libraryCard) throws Exception {
+    public LibraryCard createLibraryCard(LibraryCard libraryCard, User user) throws Exception {
         throwIfNull(libraryCard);
-        checkForValidInstitution(libraryCard.getInstitutionId());
+        Boolean isAdmin = checkIsAdmin(user);
+        //If user is not an admin, use user's institutionId rather than the value provided in the payload
+        if (!isAdmin) {
+            libraryCard.setInstitutionId(user.getInstitutionId());
+        }
```

Comment:
This is a nice check, but it might be better to throw a bad request exception here. Just thinking out loud here.